functions.go

```diff
@@ -49,6 +56,49 @@ type FunctionsNamespaceCreateRequest struct {
 	Region string `json:"region"`
 }
 
+type triggersRoot struct {
+	Triggers []FunctionsTrigger `json:"triggers,omitempty"`
+}
+
+type triggerRoot struct {
+	Trigger *FunctionsTrigger `json:"trigger,omitempty"`
+}
+
+type FunctionsTrigger struct {
+	Namespace        string                   `json:"namespace,omitempty"`
+	Function         string                   `json:"function,omitempty"`
+	Type             string                   `json:"type,omitempty"`
+	Name             string                   `json:"name,omitempty"`
+	IsEnabled        bool                     `json:"is_enabled"`
+	CreatedAt        time.Time                `json:"created_at,omitempty"`
+	UpdatedAt        time.Time                `json:"updated_at,omitempty"`
+	ScheduledDetails *TriggerScheduledDetails `json:"scheduled_details,omitempty"`
+	ScheduledRuns    *TriggerScheduledRuns    `json:"scheduled_runs,omitempty"`
+}
+
+type TriggerScheduledDetails struct {
+	Cron string                 `json:"cron,omitempty"`
+	Body map[string]interface{} `json:"body,omitempty"`
+}
+
+type TriggerScheduledRuns struct {
+	LastRunAt time.Time `json:"last_run_at,omitempty"`
+	NextRunAt time.Time `json:"next_run_at,omitempty"`
+}
+
+type FunctionsTriggerCreateRequest struct {
+	Name             string                   `json:"name"`
+	Type             string                   `json:"type"`
+	Function         string                   `json:"function"`
+	IsEnabled        bool                     `json:"is_enabled"`
+	ScheduledDetails *TriggerScheduledDetails `json:"scheduled_details,omitempty"`
+}
+
+type FunctionsTriggerUpdateRequest struct {
+	IsEnabled        bool                     `json:"is_enabled"`
```

Comment:
Should this be a pointer? Do we support updating `scheduled_details` without sending `is_enabled`? Does the API differentiate between not sending `is_enabled` and sending `is_enabled: false`? Without making this a pointer, Go will default to setting this false if excluded.

Reply:
Thanks Andrew, yes the API does allow changing `scheduled_details ` without changing `is_enabled `, I changed the `FunctionsTriggerUpdateRequest.IsEnabled` to be a `*bool`.